kong/plugin_service.go

```diff
@@ -153,6 +157,27 @@ func (s *PluginService) CreateForRoute(ctx context.Context,
 	return s.sendRequest(ctx, plugin, fmt.Sprintf("/routes/%v"+queryPath, *routeIDorName), method)
 }
 
+// CreateForConsumerGroup creates a Plugin in Kong at ConsumerGroup level.
+// If an ID is specified, it will be used to
+// create a plugin in Kong, otherwise an ID
+// is auto-generated.
+func (s *PluginService) CreateForConsumerGroup(ctx context.Context,
+	cgIDorName *string, plugin *Plugin,
+) (*Plugin, error) {
+	queryPath := "/plugins"
+	method := "POST"
+
+	if plugin.ID != nil {
```

Comment:
Can `plugin` be nil? If that's the case this will blow up 💥

Reply:
You're right. Fixed